src/proxy_handler.cc

```diff
@@ -159,6 +262,33 @@ ProxyHandler::HandleRequest(const Request& request, Response* response)
     if(status_code == 200)
       break;
   }
+  if(cache_.find(request.uri()) != cache_.end()) {
```

Comment:
The server caches whatever the website we connect to says to cache in the Cache-Control field. The websites I tested, like ucla.edu, don't say to cache the 302 response. Is the change that you are suggesting to make the server not cache 302 responses even if the website specifies that it can be cached?